src/dmd/expressionsem.d

```diff
@@ -4974,6 +4974,11 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
                 error(exp.loc, "Runtime type information is not supported for `extern(C++)` classes");
                 e = new ErrorExp();
             }
+            else if (!Type.typeinfoclass)
+            {
+                error(exp.loc, "`object.TypeInfo_Class` could not be found, but is implicitly used");
```

Comment:
Perhaps add ` by typeid` to the error message.

Reply:
It is consistent with the error message in no_TypeInfo.d, which the test is:
```
void test()
{
    int i;
    auto ti = typeid(i);
}
```
Which would be identical to this, except not getting the typeid of a class.